src/main/java/uk/gov/hmcts/jsonstore/JsonStoreWithInheritance.java

```diff
@@ -1,215 +1,220 @@
+
+    public JsonStoreWithInheritance() {
+        this("_guid_", "_extends_");
+    }
+
+    public JsonStoreWithInheritance(String idFieldName, String inheritanceFieldName) {
+        this.idFieldName = idFieldName;
+        this.inheritanceFieldName = inheritanceFieldName;
+    }
+
+    protected JsonNode getRootNode() throws Exception {
+        if (rootNode == null)
+            loadStore();
+        return rootNode;
+    }
+
+    protected Map<String, JsonNode> getNodeLibrary() {
+        if (rootNode == null)
+            loadStore();
+        return nodeLibrary;
+    }
+
+    private void loadStore() {
+        try {
+            buildObjectStore();
+            addToLibrary(rootNode);
+            for (String id : nodeLibrary.keySet())
+                inheritAndOverlayValuesFor(nodeLibrary.get(id));
+            removeInheritanceMechanismFields(rootNode);
+        } catch (Throwable t) {
+            t.printStackTrace();
+            throw new RuntimeException(t);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    public <T> Map<String, T> getMapWithIds(Class<? extends T> clazz) throws Exception {
+        Map<String, T> objectLibrary = (Map<String, T>) objectLibraryPerTypes.get(clazz);
+        if (objectLibrary == null) {
+            objectLibrary = new HashMap<String, T>();
+            Set<String> keys = getNodeLibrary().keySet();
+            ObjectMapper om = new ObjectMapper();
+            om.enable(SerializationFeature.INDENT_OUTPUT);
+            om.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+            for (String key : keys) {
+                JsonNode nodeInLibrary = nodeLibrary.get(key);
+                String jsonText = om.writeValueAsString(nodeInLibrary);
+                T anOnject = om.readValue(jsonText, clazz);
+                try {
+                    if (ReflectionUtils.retrieveFieldInObject(anOnject, idFieldName) != null) {
+                        objectLibrary.put(key, anOnject);
+                    }
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
+            objectLibraryPerTypes.put(clazz, objectLibrary);
+        }
+        return objectLibrary;
+    }
+
+    public <T> T getObjectWithId(String id, Class<? extends T> clazz) throws Exception {
+        return getMapWithIds(clazz).get(id);
+    }
+
+    private void removeInheritanceMechanismFields(JsonNode node) {
+        if (node.has(INHERITANCE_APPLIED)) {
+            ((ObjectNode) node).remove(INHERITANCE_APPLIED);
+        }
+        if (node.has(inheritanceFieldName)) {
+            ((ObjectNode) node).remove(inheritanceFieldName);
+        }
+
+        Iterator<JsonNode> fields = node.iterator();
+        while (fields.hasNext())
+            removeInheritanceMechanismFields(fields.next());
+    }
+
+    protected abstract void buildObjectStore() throws Exception;
+
+    protected void validateGUIDs(String guid) {
```

Comment:
validateGuid --> singular name is better as it's validating the 1 guid from the method param.

Reply:
Done now